src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,175 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.OrderDao;
+import com.internet.shop.exceptions.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Order;
+import com.internet.shop.model.Product;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class OrderDaoJdbcImpl implements OrderDao {
+    @Override
+    public Order create(Order order) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "INSERT INTO orders (user_id) VALUES (?);";
+            PreparedStatement statement =
+                    connection.prepareStatement(query, PreparedStatement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, order.getUserId());
+            statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            while (generatedKeys.next()) {
+                long orderId = generatedKeys.getLong(1);
+                order.setId(orderId);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("It is impossible to create a order for user id: "
+                    + order.getUserId(), e);
+        }
+        insertDataToOrdersProducts(order);
+        return order;
+    }
+
+    @Override
+    public Optional<Order> getById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM orders WHERE order_id = ? AND deleted = FALSE";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Order order = createOrderFromResultSet(resultSet);
+                order.setProducts(extractProductsForOrder(order.getId()));
```

Comment:
let's not open connection inside connection and close the previous statement before opening the new one